### PR TITLE
chore(checkoutservice): add producer interceptor for tracing

### PR DESCRIPTION
**Pull request by tranngoclam**
# Changes

Like accountingservice, this PR adds trace interceptor for producer in checkoutservice

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [x] `CHANGELOG.md` updated to document new feature additions
* [ ] Appropriate documentation updates in the [docs][]
* [ ] Appropriate Helm chart updates in the [helm-charts][]

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/
[helm-charts]: https://github.com/open-telemetry/opentelemetry-helm-charts

### CODE REVIEW

**Comment by fatsheep9146** [comment]
please add changelog

**Comment by tranngoclam** [comment]
> please add changelog

Thanks, changelog is updated

File: src/checkoutservice/kafka/trace_interceptor.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+type OTelInterceptor struct {
+	tracer     trace.Tracer
+	fixedAttrs []attribute.KeyValue
+}
+
+// NewOTelInterceptor processes span for intercepted messages and add some
+// headers with the span data.
+func NewOTelInterceptor() *OTelInterceptor {
+	oi := OTelInterceptor{}
+	oi.tracer = otel.Tracer("github.com/open-telemetry/opentelemetry-demo/checkoutservice/sarama")
```
Comment (fatsheep9146) [inline src/checkoutservice/kafka/trace_interceptor.go:27]: I thinks "github.com/IBM/sarama" is better

Comment (puckpuck) [reply src/checkoutservice/kafka/trace_interceptor.go:27]: This should be `checkoutservice` to be consistent with how we are naming the Tracer with other parts of this service: https://github.com/open-telemetry/opentelemetry-demo/blob/main/src/checkoutservice/main.go#L159

**Review by fatsheep9146 — APPROVED** [review]

**Review by puckpuck — APPROVED** [review]

**Pull request merged**